Respin of the 2.6 version that went into BK head some hours ago.

 - atomt

diff -Naur linux-2.4.26/include/asm-i386/i387.h linux-2.4.26-1/include/asm-i386/i387.h
--- linux-2.4.26/include/asm-i386/i387.h	2002-08-03 02:39:45.000000000 +0200
+++ linux-2.4.26-1/include/asm-i386/i387.h	2004-06-13 05:21:51.000000000 +0200
@@ -34,7 +34,7 @@
 
 #define clear_fpu( tsk ) do { \
 	if ( tsk->flags & PF_USEDFPU ) { \
-		asm volatile("fwait"); \
+		asm volatile("fnclex ; fwait"); \
 		tsk->flags &= ~PF_USEDFPU; \
 		stts(); \
 	} \
diff -Naur linux-2.4.26/include/asm-x86_64/i387.h linux-2.4.26-1/include/asm-x86_64/i387.h
--- linux-2.4.26/include/asm-x86_64/i387.h	2003-08-25 13:44:44.000000000 +0200
+++ linux-2.4.26-1/include/asm-x86_64/i387.h	2004-06-13 05:27:55.000000000 +0200
@@ -34,7 +34,7 @@
 
 #define clear_fpu( tsk ) do { \
 	if ( tsk->flags & PF_USEDFPU ) { \
-		asm volatile("fwait"); \
+		asm volatile("fnclex ; fwait"); \
 		tsk->flags &= ~PF_USEDFPU; \
 		stts(); \
 	} \
